strider/trapi.py

```diff
@@ -410,6 +414,92 @@ async def fill_categories_predicates(
                 node["categories"] = []
 
 
+def convert_subclasses_to_aux_graphs(
+    message: Message,
+    kp_id: str,
+    logger: logging.Logger = logging.getLogger(),
+):
+    """Take any subclass results and move them to aux graphs."""
+    for result in message.results:
+        for node_binding in result.node_bindings.values():
+            for node in node_binding:
+                if node.query_id is not None:
+                    # move query_id to id
+                    subclass = node.id
+                    node.id = node.query_id
+                    node.query_id = None
+
+                    primary_edge_id = str(uuid.uuid4())[:8]
+                    aux_edge_id = str(uuid.uuid4())[:8]
+                    subclass_edge_id = str(uuid.uuid4())[:8]
+                    original_edge_id = None
+                    original_edge = None
+
+                    # get original edge
+                    for analysis in result.analyses:
+                        for edge_bindings in analysis.edge_bindings.values():
+                            for edge_binding in edge_bindings:
+                                original_edge_id = edge_binding.id
+                                original_edge = message.knowledge_graph.edges[
+                                    original_edge_id
+                                ]
+                                # set result edge binding as one with superclass
+                                edge_binding.id = primary_edge_id
+
+                    # create subclass edge in knowledge graph
+                    message.knowledge_graph.edges[subclass_edge_id] = Edge.parse_obj(
+                        {
+                            "subject": subclass,
+                            "object": node.id,
+                            "predicate": "biolink:subclass_of",
+                            "attributes": [],
+                            "sources": [
```

Comment:
We should be adding aragorn as an aggregator knoweldge source here.

Reply:
I believe Aragorn is added later on in the Strider process, but I can confirm that Aragorn is listed:
```json
"27b4a949c1a7": {
          "subject": "CHEBI:6149",
          "object": "CHEBI:3215",
          "predicate": "biolink:subclass_of",
          "sources": [
            {
              "resource_id": "infores:automat-hetionet",
              "resource_role": "primary_knowledge_source"
            },
            {
              "resource_id": "infores:aragorn",
              "resource_role": "aggregator_knowledge_source",
              "upstream_resource_ids": ["infores:automat-hetionet"]
            }
          ],
          "attributes": []
        }
```